fabric-level-generators-v1/src/main/java/net/fabricmc/fabric/mixin/level/generator/MixinGameJoinS2CPacket.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.level.generator;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import net.minecraft.client.network.packet.GameJoinS2CPacket;
+import net.minecraft.world.level.LevelGeneratorType;
+
+@Mixin(GameJoinS2CPacket.class)
+public class MixinGameJoinS2CPacket {
+	@Redirect(method = "write", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/level/LevelGeneratorType;getName()Ljava/lang/String;"))
```

Comment:
This would be better as a field redirect

Reply:
I don't understand
How can I do that?

Reply:
`@At(value = "FIELD", target = "Lnet/minecraft/world/level/LevelGeneratorType;DEFAULT:Lnet/minecraft/world/level/LevelGeneratorType;")`

Reply:
`@At(value = "FIELD", target = "Lnet/minecraft/network/packet/s2c/play/GameJoinS2CPacket;generatorType:Lnet/minecraft/world/level/LevelGeneratorType;")`.
Annoyingly this means you won't be able to merge the mixins as I suggested in the other comment, but doing this is better.

Reply:
@Earthcomputer this `write` redirect is better because this redirect should not happen if this packet is transported locally

Reply:
They are both in `write`. It's just that my one targets a field in `write` rather than a method on the field